snuba/datasets/entities/generic_metrics.py

```diff
@@ -205,3 +223,32 @@ def __init__(self) -> None:
                 ],
             ),
         )
+
+
+class GenericMetricsStoragePartitionSelector(StoragePartitionSelector):
```

Comment:
I think this could be a little more generalized, and not specific to generic_metrics (which will be useful for me when I move this to config). 

Let's move the name "org_id" to be a constructor argument (`partition_key_column_name`) which we'll store on the object and then we can call this a `ColumnBasedStoragePartitionSelector` and put it somewhere less "generic metrics" specific. It could maybe even be in `snuba/datasets/plans/partitioned_storage.py`